keyboards/rgbkb/common/touch_encoder.c

```diff
@@ -114,10 +114,18 @@ enum {  // QT2120 registers
 
 bool     touch_initialized  = false;
 bool     touch_disabled = false;
+bool     is_swiping = false;  // tracks whether the user is currently swiping
+bool     is_holding = false;  // tracks whether the user is currently holding- I couldn't come up with a way to avoid these booleans, but I'm sure a more elegant approach exists
+//uint8_t  hold_start = 0;  // remembers the position of the start of the hold - for tracking transitions to slides //unnecessary, I think, as long as you don't update touch_processed[3]
+// without also setting is_swiping true
 uint8_t  touch_handness = 0;
 // touch_raw & touch_processed store the Detection Status, Key Status (x2), and Slider Position values
 uint8_t  touch_raw[4]       = { 0 };
 uint8_t  touch_processed[4] = { 0 };
+// JoelR note - I have interpreted touch_raw[0] as a boolean of whether or not the touchbar is currently being touched - if that assumption is wrong, then the logic I put in for holding will be wrong
```

Comment:
This is an incorrect assumption. touch_raw[0] is the detection status of the chip which is 8 bits long (1 byte) and contains multiple flags and you have to test each flag individually: CALIBRATE, OVERFLOW, SDET, and TDET
http://ww1.microchip.com/downloads/en/DeviceDoc/doc9634.pdf

Though I think most of these comments are out of date looking at the recent code.

Reply:
Ah, I see - it might have been more accurate to say that I interpreted "touch_raw[0] & SLIDER_BIT" as the above boolean. Which is perhaps the case? I never actually inserted it anywhere, I just hijacked the existing slider function, but this was how I rationalised that decision.

Reply:
I think there will always be a sensitivity trade-off between holding and swiping, but could maybe be offset a bit by adding a second TOUCH_TERM-style variable?